shopBuddy/shopBuddy/VCs/ItemDetailViewController.m

```diff
@@ -145,26 +158,41 @@ - (void) _priceSelected: (Price *)selected withCompletion:(void(^)(BOOL succeede
 #pragma mark - CollectionView
 
 - (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
+    if(doneLoading) {
+        if(self.item.prices.count==0) {
+            return 1;
+        }
+    }
     return self.item.prices.count;
 }
 
 - (__kindof UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
     PriceCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"PriceCell" forIndexPath:indexPath];
-    Price *const price = self.item.prices[indexPath.item];
-    cell.storeLabel.text = price.store;
-    cell.priceLabel.text = [price.price stringValue];
-    
+    if(doneLoading) {
+        if(self.item.prices.count>0) {
+            Price *const price = self.item.prices[indexPath.item];
+            cell.storeLabel.text = price.store;
+            cell.priceLabel.text = [NSString stringWithFormat:@"$ %.2f",[price.price doubleValue]];
+        }
+        else {
+            cell.storeLabel.text = @"No Prices Found";
+            cell.priceLabel.text = @" ";
+        }
+        cell.layer.masksToBounds = YES;
+        cell.layer.cornerRadius = 10;
+    }
     return cell;
 }
 
 - (void) collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath {
-    Price *const price = self.item.prices[indexPath.item];
-    __weak __typeof__(self) weakSelf = self;
-    [self _priceSelected:price withCompletion:^(BOOL succeeded) {
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [weakSelf performSegueWithIdentifier:@"segueFromPriceToList" sender:self];
-        });
-    }];
+    if(self.item.prices.count>0) {
+        Price *const price = self.item.prices[indexPath.item];
+        [self _priceSelected:price withCompletion:^(BOOL succeeded) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                [self.navigationController popToRootViewControllerAnimated:YES];
```

Comment:
Same, also adding to above comments this could lead to retain cycles that’s why weak ref is encouraged.